x/clp/keeper/liquidityprovider.go

```diff
@@ -71,18 +73,26 @@ func (k Keeper) DestroyLiquidityProvider(ctx sdk.Context, symbol string, lpAddre
 	store.Delete(key)
 }
 
-func (k Keeper) GetLiquidityProvidersForAsset(ctx sdk.Context, asset types.Asset) []types.LiquidityProvider {
-	var lpList []types.LiquidityProvider
+func (k Keeper) GetLiquidityProvidersForAsset(ctx sdk.Context, asset types.Asset) []*types.LiquidityProvider {
+	var lpList []*types.LiquidityProvider
 	iterator := k.GetLiquidityProviderIterator(ctx)
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
 		var lp types.LiquidityProvider
 		bytesValue := iterator.Value()
+		if !(len(bytesValue) > 0) {
+			panic("GetLiquidityProvidersForAsset ==>> len(bytesValue) <= 0")
```

Comment:
Should we be panicking here? What if there are no LPs for a given asset, shouldn't that just return an error?

Reply:
it is for testing purposes, the logic should be updated after fixing the issue